services/src/main/java/org/keycloak/connections/httpclient/ProxyMappings.java

```diff
@@ -181,6 +209,25 @@ private static HttpHost toProxyHost(String proxyUriString) {
       return new HttpHost(uri.getHost(), uri.getPort(), uri.getScheme());
     }
 
+    private static String[] toProxyCredentials(String proxyUriString) {
+
+      if (NO_PROXY.equals(proxyUriString)) {
+        return null;
+      }
+
+      URI uri = URI.create(proxyUriString);
+      String userInfo = uri.getUserInfo();
+      String[] proxyCredentials = null;
+      if (userInfo != null) {
+        if(userInfo.indexOf(":") > 0){
+          proxyCredentials = userInfo.split(":");
```

Comment:
```suggestion
          proxyCredentials = userInfo.split(":", 2);
```

Reply:
Thank you. I will add the `limit` parameter.